[Database] TAPP-130: Create new seed data & add new ER diagram

michellemtchai opened the pull request:
Please see issue #130 for more details.

To test this branch:
1. Run `docker-compose up --build`
2. Run `docker-compose exec rake db:migrate`
3. Run `docker-compose exec rake db:seed`

The last command should print out the list of tables that has been updated/inserted with new data. Furthermore, you should be able to re-run the last command several times without causing issues with the migration. The database should also update if you change the corresponding JSON files in the `/api/db/seed` folder.

jmzaleski commented:
is there some matching prose that should go in  the readme?

simpslandyy commented:
@jmzaleski I just modified the readme not too long ago in the other PR to reflect the changes in this PR :) You can go take a look!

freeatnet commented on Dockerfile:
```diff
@@ -15,13 +19,22 @@ RUN apk add --no-cache curl && \
 COPY . /srv/app/
 
+RUN if [ ${RAILS_ENV} = 'production' ]; then \
+bundle exec rake webpacker:compile; \
```
✅

freeatnet commented on dev.env.default:
```diff
@@ -1,5 +1,6 @@
 # Environment variables needed for running this project
 # copy this file to .env in this directory for it to be picked up by docker-compose
+RAILS_ENV=development
```
Oops. `COMPOSE_FILE=docker-compose.yml:docker-compose.dev.yml` got left out here :(

simpslandyy replied:
I'll fix it right away.

michellemtchai committed:
fix(db): add updated ER diagram
added new ER diagram made from current state of schema

michellemtchai committed:
feat(db): create new seed data based on old seed data
use the existing seed data as a base to create new seed data
add function that checks whether a database entry already exists in the same table, if insertion/update to a table succeed or not, etc.

michellemtchai committed:
feat(db): create seed data for new schema part 2
create seed data for the following tables:
applicant
position
applicant_data_for matching
application
assignment
offer
position_data_for_ad
position_data_for_matching
position_preference
position_template
reporting_tag
wage_chunk

update function in seed.rb to accept has_many foreign key declaration

michellemtchai committed:
fix(db): fix inconsistent indentations
make all indentation consistent

michellemtchai committed:
fix(db): remove unused json for seed data
remove preference.json
